sklearn/utils/tests/test_testing.py

```diff
@@ -491,3 +496,49 @@ def test_check_docstring_parameters():
          'type definition for param: "c " (type definition was "")',
          'sklearn.utils.tests.test_testing.f_check_param_definition There was '
          'no space between the param name and colon ("d:int")'])
+
+
+def test_assert_consistent_docs():
```

Comment:
Here we should be testing that the assert function works properly, i.e. by inventing or copying docstrings to test ordinary and tricky cases.

Tests for metric docstrings belong with metric tests, I think

Reply:
This might take some time. I will start working on it with all the other changes.